### DISCUSSION TITLE
The question about an error when test the example of the Theory of  Finite Fields.

### DISCUSSION THREAD

**Discussion by valeryjust** [cvc5/cvc5#10430] [Q&A] [answered]
When I test the example of https://cvc5.github.io/docs/cvc5-1.1.1/examples/finite_field.html.

terminate called after throwing an instance of 'cvc5::CVC5ApiException'
  what():  cvc5 can't solve field problems since it was not configured with --cocoa

But I do not find the option "--cocoa".

**Comment by martin-cs** [comment, marked as answer]
Hi @valeryjust 

CoCoA is the library we use for doing Grobner basis calculations.  It needs to be configured at compile time.  If you are building the solver by yourself then you will need to give the `--cocoa` option to `configure` before the build.  If you are using a pre-built binary, can you let us know which one as it should be built with CoCoA enabled.

**Reply by valeryjust** [reply]
Hi @martin-cs , Thank you for your answering.      

I using the  pre-built linux-static-lib.

And the version information:

This is cvc5 version 1.1.1
compiled with GCC version 9.4.0
on Feb 20 2024 21:52:16

**Reply by alex-ozdemir** [reply]
Hi @valeryjust ,

Revisiting this with a small correction: the pre-built binaries do not include CoCoA because its license is GPL. If you want to use the field solver, you'll need to build from source yourself.

We'd love to include a non-GPL computer algebra library. There is at least one [good candidate](https://github.com/lvella/polynomial-solver). Let me know if you want to work on this!

Cheers,
Alex

**Reply by daniel-larraz** [reply]
@valeryjust . Just an update: With the latest release of cvc5 ([1.2.0](https://github.com/cvc5/cvc5/releases/tag/cvc5-1.2.0)), we are now providing pre-built binaries that include CoCoA. These binaries are included in the ZIP files with the suffix `-gpl`.
